File: curtain-store/models/sku.js
/**
 * @Description Sku
 * @author coisini
 * @date Aug 30, 2021
 * @Version 1.0
 */

import {Http} from "../utils/http";

export class Sku {
    static async getSkusByIds(ids) {
        /**
         * 模板字符串将数组转换成字符串
         * ids=1,2,3,4
         */
        const res = await Http.request({
            url: `/sku?ids=${ids}`
        })
        return res
    }
}

File: curtain-manage/src/main/java/com/coisini/curtain/model/Sku.java
package com.coisini.curtain.model;

import java.math.BigDecimal;
import java.util.Date;
import java.util.List;

import com.baomidou.mybatisplus.annotation.*;
import com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;

/**
 * @Description Sku
 * @author coisini
 * @date Sep 17, 2021
 * @Version 1.0
 */
@Data
@TableName(value = "sku", autoResultMap = true)
public class Sku {

    @TableId(value = "id", type = IdType.AUTO)
    private Integer id;

    @JsonIgnore
    private Date createTime;

    @JsonIgnore
    private Date updateTime;

    @TableLogic
    @JsonIgnore
    private Date deleteTime;

    private BigDecimal price;

    private BigDecimal discountPrice;

    private Integer online;

    private String img;

    private String title;

    private Integer spuId;

    @TableField(typeHandler = JacksonTypeHandler.class)
    private List<SpecKeyValue> specs;

    private String code;

    private Integer stock;

    private Integer categoryId;

    private Integer rootCategoryId;

    private String test;


}
